src/deepEqual.js

```diff
@@ -19,8 +19,27 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5}}) === true
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  */
-function deepEqual(a, b) {
-  // write code here
+function deepEqual(obj1, obj2) {
+  if (obj1 === obj2) {
+    return true;
+  }
+
+  if (obj1 === null || typeof (obj2) !== 'object') {
+    return false;
+  }
+
+  var numPropertiesInObj1 = 0;
+  var numPropertiesInObj2 = 0;
+  for (var property in obj1) {
```

Comment:
do you need this loop?

Reply:
Sorry, I've tried in different variations ), but I don't understand, what exactly you mean). It have not passed the test without loop (if I change it for if statement,for example). Would you please explain  in more detail?

Reply:
I leave a working solution because I need to move one with other tasks). Commented out the solution that we discussed in the lesson in the morning. I would be grateful if you explain exactly how you proposed to do).

Reply:
This solution looks good. If I wanted changes to this solution - I must have been too picky :) 

Unfortunately, I don't remember the previous state of the code. I think I wanted you to get rid of "numPropertiesInObj2 += 1;".